Sources/Benchmark/BenchmarkThresholds.swift

```diff
@@ -32,3 +32,175 @@ public struct BenchmarkThresholds: Codable {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+
+            let base = try container.decodeIfPresent(Int.self, forKey: .base)
+            let tolerancePercentage = try container.decodeIfPresent(Double.self, forKey: .tolerancePercentage)
+            let min = try container.decodeIfPresent(Int.self, forKey: .min)
+            let max = try container.decodeIfPresent(Int.self, forKey: .max)
+
+            var relative: Relative?
+            var range: Range?
+
+            if let base, let tolerancePercentage {
+                relative = Relative(base: base, tolerancePercentage: tolerancePercentage)
+
+                guard base > 0, tolerancePercentage > 0 else {
+                    throw DecodingError.dataCorrupted(
+                        .init(
+                            codingPath: decoder.codingPath,
+                            debugDescription: """
+                                RelativeOrRange thresholds object contains an invalid relative values.
+                                'base' (\(base)) and 'tolerancePercentage' (\(tolerancePercentage)) must be positive.
+                                """
+                        )
+                    )
+                }
+            }
+            if let min, let max {
+                range = Range(min: min, max: max)
+
+                guard min <= max else {
+                    throw DecodingError.dataCorrupted(
+                        .init(
+                            codingPath: decoder.codingPath,
+                            debugDescription: """
+                                RelativeOrRange thresholds object contains invalid min-max values.
+                                'min' (\(min)) and max ('\(max)') don't satisfy the requirements of min <= max.
+                                """
+                        )
+                    )
+                }
+            }
+
+            self.relative = relative
+            self.range = range
+
+            if !self.containsAnyValue {
+                throw DecodingError.dataCorrupted(
+                    .init(
+                        codingPath: decoder.codingPath,
+                        debugDescription: """
+                            RelativeOrRange thresholds object does not contain either a valid relative or range.
+                            For relative thresholds, both 'base' (Int) and 'tolerancePercentage' (Double) must be present and valid.
+                            For range thresholds, both 'min' (Int) and 'max' (Int) must be present and valid.
+                            You can declare both relative and range in the same object together, or just one of them.
+                            Example: { "min": 90, "max": 110 }
+                            Example: { "base": 115, "tolerancePercentage": 5.5 }
+                            Example: { "base": 115, "tolerancePercentage": 4.5, "min": 90, "max": 110 }
+                            """
```

Comment:
Some info.

This is also backward-compatible. It'll try to read an `Int` first and if it fails then it tries to read the new `RelativeOrRange`.